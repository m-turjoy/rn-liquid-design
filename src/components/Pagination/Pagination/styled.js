import styled from 'styled-components'

import {
  width,
  height,
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  color,
  paddingLeft,
  paddingRight,
  opacity,
  top
} from 'styled-system'

const InnerWrapper = styled.View`
  ${width}
  align-items: center;
  justify-content: space-between;
`

const TextWrapper = styled.Text`
  ${fontFamily}
  ${fontWeight}
  ${fontSize}
  ${lineHeight}
  ${color}
  ${opacity}
  ${top}
  text-align: center;
`

const PaginationDotWrapper = styled.View`
  ${paddingLeft}
  ${paddingRight}
  align-items: center;
  justify-content: center;
`

const PaginationContainer = styled.View`
  ${width}
  ${height}
  flex-direction: row;
  align-items: center;
  justify-content: center;
`

const IconWrapper = styled.View`
  ${width}
  ${height}
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`

export {
  InnerWrapper,
  TextWrapper,
  PaginationDotWrapper,
  PaginationContainer,
  IconWrapper
}
