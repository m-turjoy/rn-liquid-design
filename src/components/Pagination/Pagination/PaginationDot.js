import React, { PureComponent } from 'react';
import { Animated, Platform, TouchableWithoutFeedback } from 'react-native';
import { bool, number, object, string, func } from 'prop-types';
import { fonts } from '../../../config';
import { TextWrapper, PaginationDotWrapper } from './styled';

class PaginationDot extends PureComponent {
  render() {
    const {
      active,
      carouselRef,
      index,
      fontSize,
      fontFamily,
      fontWeight,
      lineHeight,
      inactiveTextColor,
      activeTextColor,
      activeBackgroundColor,
      itemWidth,
      itemHeight,
      itemPaddingLeft,
      itemPaddingRight,
      activeFontWeight,
      disabled,
      onDotPress,
    } = this.props;

    const onPress = () => {
      if (onDotPress) {
        onDotPress(index);
      }
      setTimeout(
        () =>
          carouselRef &&
          carouselRef._snapToItem(carouselRef._getPositionIndex(index)),
        100
      );
    };

    const dotStyle = {
      backgroundColor: active ? activeBackgroundColor : null,
      width: itemWidth,
      height: itemHeight,
      borderRadius: 6,
      overflow: 'hidden',
    };

    return (
      <PaginationDotWrapper
        paddingLeft={itemPaddingLeft}
        paddingRight={itemPaddingRight}
        opacity={disabled ? 0.5 : 1}
      >
        <TouchableWithoutFeedback onPress={disabled ? null : onPress}>
          <Animated.View style={dotStyle}>
            <TextWrapper
              fontFamily={
                fontFamily !== fonts.Regular
                  ? fontFamily
                  : active
                    ? fonts.Black
                    : fonts.Regular
              }
              fontSize={fontSize}
              fontWeight={active ? activeFontWeight : fontWeight}
              lineHeight={lineHeight}
              color={active ? activeTextColor : inactiveTextColor}
              top={Platform.OS === 'android' ? -5 : -1}
            >
              {index + 1}
            </TextWrapper>
          </Animated.View>
        </TouchableWithoutFeedback>
      </PaginationDotWrapper>
    );
  }
}

PaginationDot.propTypes = {
  active: bool,
  index: number,
  fontFamily: string,
  fontSize: number,
  fontWeight: number,
  lineHeight: number,
  activeTextColor: string,
  inactiveTextColor: string,
  activeBackgroundColor: string,
  itemWidth: number,
  itemHeight: number,
  itemPaddingLeft: number,
  itemPaddingRight: number,
  carouselRef: object,
  activeFontWeight: number,
  disabled: bool,
  onDotPress: func,
};

export default PaginationDot;
