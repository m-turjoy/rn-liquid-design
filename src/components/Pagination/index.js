import React, { Component } from 'react'
import {
  View,
  Dimensions
} from 'react-native'
import {
  bool,
  number,
  object,
  func,
  array,
  string,
  oneOfType,
  shape
} from 'prop-types'
import { ThemeProvider } from 'styled-components'
import {
  fonts,
  colors
} from '../../config'
import Carousel from './Carousel/Carousel'
import Pagination from './Pagination/Pagination'
import { defaultThemeName, getThemeObject } from '../../config/theme'

const { width } = Dimensions.get('window')

class PaginationCarousel extends Component {
  componentDidMount = () => {
    this.forceUpdate()
  }
  render() {
    const {
      dotsLength,
      activeDotIndex,
      carouselRef,
      windowSize,
      data,
      renderItem,
      onSnapToItem,
      refProp,
      fontFamily,
      fontSize,
      fontWeight,
      activeFontWeight,
      lineHeight,
      activeTextColor,
      inactiveTextColor,
      activeBackgroundColor,
      itemWidth,
      itemHeight,
      itemPaddingLeft,
      itemPaddingRight,
      paginationWidth,
      paginationHeight,
      innerWidth,
      iconPaddingLeft,
      iconPaddingRight,
      sliderHeight,
      disabled,
      themeName,
      iconSize
    } = this.props
    const themeObj = getThemeObject(themeName)
    const themeColor = themeObj.colors.primary.base
    const bgColor = activeBackgroundColor || themeColor

    const pagination =
      <Pagination
        dotsLength={dotsLength}
        activeDotIndex={activeDotIndex}
        carouselRef={carouselRef}
        fontFamily={fontFamily}
        fontSize={fontSize}
        fontWeight={fontWeight}
        activeFontWeight={activeFontWeight}
        lineHeight={lineHeight}
        activeTextColor={activeTextColor}
        inactiveTextColor={inactiveTextColor}
        activeBackgroundColor={disabled ? colors.sensitiveGreyDarker : bgColor}
        itemWidth={itemWidth}
        itemHeight={itemHeight}
        itemPaddingLeft={itemPaddingLeft}
        itemPaddingRight={itemPaddingRight}
        paginationWidth={paginationWidth}
        paginationHeight={paginationHeight}
        innerWidth={innerWidth}
        iconPaddingLeft={iconPaddingLeft}
        iconPaddingRight={iconPaddingRight}
        disabled={disabled}
        iconColor={bgColor}
        iconSize={iconSize}
      />

    return (
      <ThemeProvider
        theme={themeObj}
      >
        <View>
          <Carousel
            windowSize={windowSize}
            ref={refProp}
            sliderWidth={width}
            sliderHeight={sliderHeight - paginationHeight}
            itemWidth={width}
            data={data}
            renderItem={renderItem}
            onSnapToItem={onSnapToItem}
            scrollEnabled={!disabled}
            useScrollView
          />
          {pagination}
        </View>
      </ThemeProvider>

    )
  }
}

PaginationCarousel.propTypes = {
  activeDotIndex: number.isRequired,
  dotsLength: number.isRequired,
  carouselRef: object,
  windowSize: number,
  data: array.isRequired,
  renderItem: func.isRequired,
  onSnapToItem: func.isRequired,
  refProp: func.isRequired,
  innerWidth: number,
  paginationWidth: oneOfType([
    string,
    number
  ]),
  paginationHeight: number,
  iconPaddingLeft: number,
  iconPaddingRight: number,
  sliderHeight: number,
  fontFamily: string,
  fontSize: number,
  fontWeight: number,
  lineHeight: number,
  activeTextColor: string,
  inactiveTextColor: string,
  activeBackgroundColor: string,
  itemWidth: number,
  itemHeight: number,
  itemPaddingLeft: number,
  itemPaddingRight: number,
  activeFontWeight: number,
  disabled: bool,
  iconSize: number,
  themeName: oneOfType([
    string,
    shape({
      primary: shape({
        lightest: string,
        light: string,
        base: string,
        dark: string,
        darker: string
      }).isRequired,
      secondary: shape({
        lightest: string,
        light: string,
        base: string,
        dark: string,
        darker: string
      }).isRequired
    })
  ])
}

PaginationCarousel.defaultProps = {
  innerWidth: width < 350 ? 130 : 210,
  paginationWidth: '100%',
  paginationHeight: 100,
  iconPaddingLeft: width === 360 ? 6 : 10,
  iconPaddingRight: width === 360 ? 6 : 10,
  sliderHeight: 500,
  fontFamily: fonts.Regular,
  fontSize: 16,
  lineHeight: 26.5,
  activeTextColor: colors.white,
  inactiveTextColor: colors.richBlackDefault,
  itemWidth: 26,
  itemHeight: 26,
  itemPaddingLeft: 8,
  itemPaddingRight: 8,
  activeFontWeight: null,
  disabled: false,
  themeName: defaultThemeName,
  iconSize: 24
}

export default PaginationCarousel
