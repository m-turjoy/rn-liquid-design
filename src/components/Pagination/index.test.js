import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from './index';

configure({ adapter: new Adapter() });
describe('Pagination component', () => {
  describe('Renders correctly', () => {
    test('it renders simple Pagination', () => {
      const tree = shallow(<Pagination />);
      expect(tree).toMatchSnapshot();
    });
  });
});
