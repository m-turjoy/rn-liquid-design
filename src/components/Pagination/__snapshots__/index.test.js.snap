// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Renders correctly it renders simple Pagination 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationCarousel
    activeFontWeight={null}
    activeTextColor="#ffffff"
    disabled={false}
    fontFamily="Lato-Regular"
    fontSize={16}
    iconPaddingLeft={10}
    iconPaddingRight={10}
    iconSize={24}
    inactiveTextColor="#1b1b25"
    innerWidth={210}
    itemHeight={26}
    itemPaddingLeft={8}
    itemPaddingRight={8}
    itemWidth={26}
    lineHeight={26.5}
    paginationHeight={100}
    paginationWidth="100%"
    sliderHeight={500}
    themeName="vibrantCyan"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View>
        <Carousel
          activeAnimationOptions={null}
          activeAnimationType="timing"
          activeSlideAlignment="center"
          activeSlideOffset={0}
          apparitionDelay={0}
          callbackOffsetMargin={1}
          containerCustomStyle={Object {}}
          contentContainerCustomStyle={Object {}}
          enableMomentum={false}
          enableSnap={true}
          firstItem={0}
          hasParallaxImages={false}
          inactiveSlideOpacity={0.7}
          inactiveSlideScale={0.9}
          inactiveSlideShift={0}
          itemWidth={750}
          layout="default"
          lockScrollTimeoutDuration={1000}
          lockScrollWhileSnapping={false}
          loop={false}
          loopClonesPerSide={3}
          scrollEnabled={true}
          shouldOptimizeUpdates={true}
          slideStyle={Object {}}
          sliderHeight={400}
          sliderWidth={750}
          swipeThreshold={20}
          useScrollView={true}
        />
        <Pagination
          activeBackgroundColor="#2dbecd"
          activeFontWeight={null}
          activeTextColor="#ffffff"
          disabled={false}
          fontFamily="Lato-Regular"
          fontSize={16}
          iconColor="#2dbecd"
          iconPaddingLeft={10}
          iconPaddingRight={10}
          iconSize={24}
          inactiveTextColor="#1b1b25"
          innerWidth={210}
          itemHeight={26}
          itemPaddingLeft={8}
          itemPaddingRight={8}
          itemWidth={26}
          lineHeight={26.5}
          paginationHeight={100}
          paginationWidth="100%"
        />
      </View>,
      "theme": Object {
        "colors": Object {
          "black": Object {
            "base": undefined,
          },
          "primary": Object {
            "base": "#2dbecd",
            "dark": "#29abb9",
            "darker": "#208590",
            "light": "#57cbd7",
            "lightest": "#abe5eb",
          },
          "richBlack": Object {
            "base": undefined,
            "dark": undefined,
            "darker": undefined,
            "darkest": undefined,
            "light": undefined,
            "lighter": undefined,
            "lightest": undefined,
          },
          "richBlue": Object {
            "base": "#0f69af",
            "dark": "#0e5f9e",
            "darker": "#0b4a7b",
            "darkest": "#083558",
            "light": "#3f87bf",
            "lighter": "#6fa5cf",
            "lightest": "#9fc3df",
          },
          "richGreen": Object {
            "base": "#149b5f",
            "dark": "#0d6c42",
            "darker": "#0c5d39",
            "darkest": "#094d2f",
            "light": "#5ab98f",
            "lighter": "#72c39f",
            "lightest": "#89cdaf",
          },
          "richPurple": Object {
            "base": "#503291",
            "dark": "#482d83",
            "darker": "#382366",
            "darkest": "#281949",
            "light": "#735ba7",
            "lighter": "#9684bd",
            "lightest": "#b9add3",
          },
          "richRed": Object {
            "base": "#e61e50",
            "dark": "#cf1b48",
            "darker": "#a11538",
            "darkest": "#730f28",
            "light": "#eb4b73",
            "lighter": "#f07896",
            "lightest": "#f5a5b9",
          },
          "secondary": Object {
            "base": "#ffc832",
            "dark": "#f3be2f",
            "darker": "#ddac28",
            "light": "#ffd35b",
            "lightest": "#ffe9ad",
          },
          "sensitiveGrey": Object {
            "base": "#f8f8fc",
            "dark": "#f3f3f7",
            "darker": "#efeff1",
            "darkest": "#e9e9ed",
            "light": "#f8f8fc",
            "lighter": "#ffffff",
            "lightest": "#ffffff",
          },
          "vibrantCyan": Object {
            "base": "#2dbecd",
            "dark": "#29abb9",
            "darker": "#208590",
            "darkest": "#175f67",
            "light": "#57cbd7",
            "lighter": "#81d8e1",
            "lightest": "#abe5eb",
          },
          "vibrantGreen": Object {
            "base": undefined,
            "dark": undefined,
            "darker": undefined,
            "darkest": undefined,
            "light": undefined,
            "lighter": undefined,
            "lightest": undefined,
          },
          "vibrantMagenta": Object {
            "base": "#eb3c96",
            "dark": "#d43687",
            "darker": "#a52a69",
            "darkest": "#761e4b",
            "light": "#ef63ab",
            "lighter": "#f38ac0",
            "lightest": "#f7b1d5",
          },
          "vibrantYellow": Object {
            "base": "#ffc832",
            "dark": "#f3be2f",
            "darker": "#ddac28",
            "darkest": "#806419",
            "light": "#ffd35b",
            "lighter": "#ffde84",
            "lightest": "#ffe9ad",
          },
          "white": Object {
            "base": "#ffffff",
          },
        },
        "fontSizes": Array [
          8,
          10,
          12,
          14,
          16,
          18,
          20,
          24,
          36,
        ],
        "fontWeights": Array [
          100,
          200,
          300,
          400,
          500,
          600,
          700,
          800,
          900,
        ],
        "fonts": Object {
          "Black": "Lato-Black",
          "Bold": "Lato-Bold",
          "IcoMoon": "icomoon",
          "Merck": "MerckWeb-Regular",
          "Regular": "Lato-Regular",
        },
        "heights": Object {
          "Big": 50,
          "Default": 40,
        },
        "shadows": Object {
          "active": Object {
            "elevation": 40,
            "shadowColor": "#1b1b25",
            "shadowOffset": "0px 30px",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
          },
          "default": Object {},
          "stacked": Object {
            "elevation": 25,
            "shadowColor": "#1b1b25",
            "shadowOffset": "0px 30px",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
          },
        },
        "space": Array [
          0,
          1,
          2,
          5,
          10,
          15,
          20,
          24,
          50,
          8,
          7,
          4,
          9,
          6,
        ],
        "stacked": Object {
          "stackedFirst": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "position": "absolute",
            "zIndex": 10,
          },
          "stackedSecond": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "left": 3,
            "position": "absolute",
            "top": 3,
            "zIndex": 9,
          },
          "stackedThird": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "left": 6,
            "top": 6,
            "zIndex": 8,
          },
        },
        "widths": Array [
          50,
          100,
          130,
        ],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Carousel
            activeAnimationOptions={null}
            activeAnimationType="timing"
            activeSlideAlignment="center"
            activeSlideOffset={0}
            apparitionDelay={0}
            callbackOffsetMargin={1}
            containerCustomStyle={Object {}}
            contentContainerCustomStyle={Object {}}
            enableMomentum={false}
            enableSnap={true}
            firstItem={0}
            hasParallaxImages={false}
            inactiveSlideOpacity={0.7}
            inactiveSlideScale={0.9}
            inactiveSlideShift={0}
            itemWidth={750}
            layout="default"
            lockScrollTimeoutDuration={1000}
            lockScrollWhileSnapping={false}
            loop={false}
            loopClonesPerSide={3}
            scrollEnabled={true}
            shouldOptimizeUpdates={true}
            slideStyle={Object {}}
            sliderHeight={400}
            sliderWidth={750}
            swipeThreshold={20}
            useScrollView={true}
          />,
          <Pagination
            activeBackgroundColor="#2dbecd"
            activeFontWeight={null}
            activeTextColor="#ffffff"
            disabled={false}
            fontFamily="Lato-Regular"
            fontSize={16}
            iconColor="#2dbecd"
            iconPaddingLeft={10}
            iconPaddingRight={10}
            iconSize={24}
            inactiveTextColor="#1b1b25"
            innerWidth={210}
            itemHeight={26}
            itemPaddingLeft={8}
            itemPaddingRight={8}
            itemWidth={26}
            lineHeight={26.5}
            paginationHeight={100}
            paginationWidth="100%"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeAnimationOptions": null,
            "activeAnimationType": "timing",
            "activeSlideAlignment": "center",
            "activeSlideOffset": 0,
            "apparitionDelay": 0,
            "callbackOffsetMargin": 1,
            "containerCustomStyle": Object {},
            "contentContainerCustomStyle": Object {},
            "data": undefined,
            "enableMomentum": false,
            "enableSnap": true,
            "firstItem": 0,
            "hasParallaxImages": false,
            "inactiveSlideOpacity": 0.7,
            "inactiveSlideScale": 0.9,
            "inactiveSlideShift": 0,
            "itemWidth": 750,
            "layout": "default",
            "lockScrollTimeoutDuration": 1000,
            "lockScrollWhileSnapping": false,
            "loop": false,
            "loopClonesPerSide": 3,
            "onSnapToItem": undefined,
            "renderItem": undefined,
            "scrollEnabled": true,
            "shouldOptimizeUpdates": true,
            "slideStyle": Object {},
            "sliderHeight": 400,
            "sliderWidth": 750,
            "swipeThreshold": 20,
            "useScrollView": true,
            "windowSize": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeBackgroundColor": "#2dbecd",
            "activeDotIndex": undefined,
            "activeFontWeight": null,
            "activeTextColor": "#ffffff",
            "carouselRef": undefined,
            "disabled": false,
            "dotsLength": undefined,
            "fontFamily": "Lato-Regular",
            "fontSize": 16,
            "fontWeight": undefined,
            "iconColor": "#2dbecd",
            "iconPaddingLeft": 10,
            "iconPaddingRight": 10,
            "iconSize": 24,
            "inactiveTextColor": "#1b1b25",
            "innerWidth": 210,
            "itemHeight": 26,
            "itemPaddingLeft": 8,
            "itemPaddingRight": 8,
            "itemWidth": 26,
            "lineHeight": 26.5,
            "paginationHeight": 100,
            "paginationWidth": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View>
          <Carousel
            activeAnimationOptions={null}
            activeAnimationType="timing"
            activeSlideAlignment="center"
            activeSlideOffset={0}
            apparitionDelay={0}
            callbackOffsetMargin={1}
            containerCustomStyle={Object {}}
            contentContainerCustomStyle={Object {}}
            enableMomentum={false}
            enableSnap={true}
            firstItem={0}
            hasParallaxImages={false}
            inactiveSlideOpacity={0.7}
            inactiveSlideScale={0.9}
            inactiveSlideShift={0}
            itemWidth={750}
            layout="default"
            lockScrollTimeoutDuration={1000}
            lockScrollWhileSnapping={false}
            loop={false}
            loopClonesPerSide={3}
            scrollEnabled={true}
            shouldOptimizeUpdates={true}
            slideStyle={Object {}}
            sliderHeight={400}
            sliderWidth={750}
            swipeThreshold={20}
            useScrollView={true}
          />
          <Pagination
            activeBackgroundColor="#2dbecd"
            activeFontWeight={null}
            activeTextColor="#ffffff"
            disabled={false}
            fontFamily="Lato-Regular"
            fontSize={16}
            iconColor="#2dbecd"
            iconPaddingLeft={10}
            iconPaddingRight={10}
            iconSize={24}
            inactiveTextColor="#1b1b25"
            innerWidth={210}
            itemHeight={26}
            itemPaddingLeft={8}
            itemPaddingRight={8}
            itemWidth={26}
            lineHeight={26.5}
            paginationHeight={100}
            paginationWidth="100%"
          />
        </View>,
        "theme": Object {
          "colors": Object {
            "black": Object {
              "base": undefined,
            },
            "primary": Object {
              "base": "#2dbecd",
              "dark": "#29abb9",
              "darker": "#208590",
              "light": "#57cbd7",
              "lightest": "#abe5eb",
            },
            "richBlack": Object {
              "base": undefined,
              "dark": undefined,
              "darker": undefined,
              "darkest": undefined,
              "light": undefined,
              "lighter": undefined,
              "lightest": undefined,
            },
            "richBlue": Object {
              "base": "#0f69af",
              "dark": "#0e5f9e",
              "darker": "#0b4a7b",
              "darkest": "#083558",
              "light": "#3f87bf",
              "lighter": "#6fa5cf",
              "lightest": "#9fc3df",
            },
            "richGreen": Object {
              "base": "#149b5f",
              "dark": "#0d6c42",
              "darker": "#0c5d39",
              "darkest": "#094d2f",
              "light": "#5ab98f",
              "lighter": "#72c39f",
              "lightest": "#89cdaf",
            },
            "richPurple": Object {
              "base": "#503291",
              "dark": "#482d83",
              "darker": "#382366",
              "darkest": "#281949",
              "light": "#735ba7",
              "lighter": "#9684bd",
              "lightest": "#b9add3",
            },
            "richRed": Object {
              "base": "#e61e50",
              "dark": "#cf1b48",
              "darker": "#a11538",
              "darkest": "#730f28",
              "light": "#eb4b73",
              "lighter": "#f07896",
              "lightest": "#f5a5b9",
            },
            "secondary": Object {
              "base": "#ffc832",
              "dark": "#f3be2f",
              "darker": "#ddac28",
              "light": "#ffd35b",
              "lightest": "#ffe9ad",
            },
            "sensitiveGrey": Object {
              "base": "#f8f8fc",
              "dark": "#f3f3f7",
              "darker": "#efeff1",
              "darkest": "#e9e9ed",
              "light": "#f8f8fc",
              "lighter": "#ffffff",
              "lightest": "#ffffff",
            },
            "vibrantCyan": Object {
              "base": "#2dbecd",
              "dark": "#29abb9",
              "darker": "#208590",
              "darkest": "#175f67",
              "light": "#57cbd7",
              "lighter": "#81d8e1",
              "lightest": "#abe5eb",
            },
            "vibrantGreen": Object {
              "base": undefined,
              "dark": undefined,
              "darker": undefined,
              "darkest": undefined,
              "light": undefined,
              "lighter": undefined,
              "lightest": undefined,
            },
            "vibrantMagenta": Object {
              "base": "#eb3c96",
              "dark": "#d43687",
              "darker": "#a52a69",
              "darkest": "#761e4b",
              "light": "#ef63ab",
              "lighter": "#f38ac0",
              "lightest": "#f7b1d5",
            },
            "vibrantYellow": Object {
              "base": "#ffc832",
              "dark": "#f3be2f",
              "darker": "#ddac28",
              "darkest": "#806419",
              "light": "#ffd35b",
              "lighter": "#ffde84",
              "lightest": "#ffe9ad",
            },
            "white": Object {
              "base": "#ffffff",
            },
          },
          "fontSizes": Array [
            8,
            10,
            12,
            14,
            16,
            18,
            20,
            24,
            36,
          ],
          "fontWeights": Array [
            100,
            200,
            300,
            400,
            500,
            600,
            700,
            800,
            900,
          ],
          "fonts": Object {
            "Black": "Lato-Black",
            "Bold": "Lato-Bold",
            "IcoMoon": "icomoon",
            "Merck": "MerckWeb-Regular",
            "Regular": "Lato-Regular",
          },
          "heights": Object {
            "Big": 50,
            "Default": 40,
          },
          "shadows": Object {
            "active": Object {
              "elevation": 40,
              "shadowColor": "#1b1b25",
              "shadowOffset": "0px 30px",
              "shadowOpacity": 0.3,
              "shadowRadius": 20,
            },
            "default": Object {},
            "stacked": Object {
              "elevation": 25,
              "shadowColor": "#1b1b25",
              "shadowOffset": "0px 30px",
              "shadowOpacity": 0.3,
              "shadowRadius": 20,
            },
          },
          "space": Array [
            0,
            1,
            2,
            5,
            10,
            15,
            20,
            24,
            50,
            8,
            7,
            4,
            9,
            6,
          ],
          "stacked": Object {
            "stackedFirst": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "position": "absolute",
              "zIndex": 10,
            },
            "stackedSecond": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "left": 3,
              "position": "absolute",
              "top": 3,
              "zIndex": 9,
            },
            "stackedThird": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "left": 6,
              "top": 6,
              "zIndex": 8,
            },
          },
          "widths": Array [
            50,
            100,
            130,
          ],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Carousel
              activeAnimationOptions={null}
              activeAnimationType="timing"
              activeSlideAlignment="center"
              activeSlideOffset={0}
              apparitionDelay={0}
              callbackOffsetMargin={1}
              containerCustomStyle={Object {}}
              contentContainerCustomStyle={Object {}}
              enableMomentum={false}
              enableSnap={true}
              firstItem={0}
              hasParallaxImages={false}
              inactiveSlideOpacity={0.7}
              inactiveSlideScale={0.9}
              inactiveSlideShift={0}
              itemWidth={750}
              layout="default"
              lockScrollTimeoutDuration={1000}
              lockScrollWhileSnapping={false}
              loop={false}
              loopClonesPerSide={3}
              scrollEnabled={true}
              shouldOptimizeUpdates={true}
              slideStyle={Object {}}
              sliderHeight={400}
              sliderWidth={750}
              swipeThreshold={20}
              useScrollView={true}
            />,
            <Pagination
              activeBackgroundColor="#2dbecd"
              activeFontWeight={null}
              activeTextColor="#ffffff"
              disabled={false}
              fontFamily="Lato-Regular"
              fontSize={16}
              iconColor="#2dbecd"
              iconPaddingLeft={10}
              iconPaddingRight={10}
              iconSize={24}
              inactiveTextColor="#1b1b25"
              innerWidth={210}
              itemHeight={26}
              itemPaddingLeft={8}
              itemPaddingRight={8}
              itemWidth={26}
              lineHeight={26.5}
              paginationHeight={100}
              paginationWidth="100%"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeAnimationOptions": null,
              "activeAnimationType": "timing",
              "activeSlideAlignment": "center",
              "activeSlideOffset": 0,
              "apparitionDelay": 0,
              "callbackOffsetMargin": 1,
              "containerCustomStyle": Object {},
              "contentContainerCustomStyle": Object {},
              "data": undefined,
              "enableMomentum": false,
              "enableSnap": true,
              "firstItem": 0,
              "hasParallaxImages": false,
              "inactiveSlideOpacity": 0.7,
              "inactiveSlideScale": 0.9,
              "inactiveSlideShift": 0,
              "itemWidth": 750,
              "layout": "default",
              "lockScrollTimeoutDuration": 1000,
              "lockScrollWhileSnapping": false,
              "loop": false,
              "loopClonesPerSide": 3,
              "onSnapToItem": undefined,
              "renderItem": undefined,
              "scrollEnabled": true,
              "shouldOptimizeUpdates": true,
              "slideStyle": Object {},
              "sliderHeight": 400,
              "sliderWidth": 750,
              "swipeThreshold": 20,
              "useScrollView": true,
              "windowSize": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeBackgroundColor": "#2dbecd",
              "activeDotIndex": undefined,
              "activeFontWeight": null,
              "activeTextColor": "#ffffff",
              "carouselRef": undefined,
              "disabled": false,
              "dotsLength": undefined,
              "fontFamily": "Lato-Regular",
              "fontSize": 16,
              "fontWeight": undefined,
              "iconColor": "#2dbecd",
              "iconPaddingLeft": 10,
              "iconPaddingRight": 10,
              "iconSize": 24,
              "inactiveTextColor": "#1b1b25",
              "innerWidth": 210,
              "itemHeight": 26,
              "itemPaddingLeft": 8,
              "itemPaddingRight": 8,
              "itemWidth": 26,
              "lineHeight": 26.5,
              "paginationHeight": 100,
              "paginationWidth": "100%",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
